### ISSUE TITLE
Testing out of a git repository fails on 0.20

### ISSUE THREAD

**Issue by rdoursenaud** [Gandi/gandi.cli#215]
Testing while packaging (for Arch Linux's AUR) fails:

[2017-07-11-makepkg.log.txt](https://github.com/Gandi/gandi.cli/files/1139684/2017-07-11-makepkg.log.txt)

IMHO, the git repository shouldn't be needed to allow running the tests.

I've disabled the tests in the package for now.

**Comment by sayoun** [comment]
We need a git repository to test the `paas deploy` commands and check for regressions. The git repository used is temporary and only created for the duration of tests so I don't understand why it breaks on Arch, maybe we don't have permissions to create a temporary directory during the packaging ?

It uses Click file system isolation http://click.pocoo.org/5/testing/#file-system-isolation

**Comment by koobs** [comment]
I am seeing identical failures testing the 0.21 PyPI sdist during FreeBSD Ports QA:

Error:
```
Traceback (most recent call last):
  File "/usr/home/koobs/repos/freebsd/ports/www/py-gandi.cli/work/gandi.cli-0.21/gandi/cli/tests/commands/test_paas.py", line 343, in test_deploy_remote
    """)
AssertionError: u"Error: Could not find git remote to extract deploy url from.\nThis usually hap [truncated]... != u"ssh 185290@git.dc2.gpaas.net 'deploy default.git master'\n"
+ ssh 185290@git.dc2.gpaas.net 'deploy default.git master'
- Error: Could not find git remote to extract deploy url from.
- This usually happens when:
<snip>
```

For the following tests:

```
FAIL: test_deploy (gandi.cli.tests.commands.test_paas.PaasTestCase)
FAIL: test_deploy_guess_remote_with_branch (gandi.cli.tests.commands.test_paas.PaasTestCase)
FAIL: test_deploy_invalid_remote_content (gandi.cli.tests.commands.test_paas.PaasTestCase)
FAIL: test_deploy_remote (gandi.cli.tests.commands.test_paas.PaasTestCase)

Earlier test output shows:

```
test_delete_unknown (gandi.cli.tests.commands.test_paas.PaasTestCase) ... ok
test_deploy (gandi.cli.tests.commands.test_paas.PaasTestCase) ... fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
FAIL
test_deploy_guess_remote_with_branch (gandi.cli.tests.commands.test_paas.PaasTestCase) ... fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
FAIL
test_deploy_invalid_remote_content (gandi.cli.tests.commands.test_paas.PaasTestCase) ... fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
FAIL
test_deploy_invalid_remote_empty (gandi.cli.tests.commands.test_paas.PaasTestCase) ... fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
ok
test_deploy_remote (gandi.cli.tests.commands.test_paas.PaasTestCase) ... fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
fatal: --local can only be used inside a git repository
FAIL
test_info (gandi.cli.tests.commands.test_paas.PaasTestCase) ... ok
```

Git version is: git version 2.14.1

**Comment by sayoun** [comment]
Hello,

I'm working on this, Travis also fail on the same tests, should be fixed today.

**Comment by koobs** [comment]
See Also:

* https://github.com/arialdomartini/oh-my-git/issues/110
* http://lists-archives.com/git/895194-config-complain-about-local-outside-of-a-git-repo.html

**Comment by koobs** [comment]
Making the tests robusts to any error (skipping the test) rather than just this specific one would be great.

**Comment by sayoun** [comment]
No the tests should not be skipped, we need to test the deploy command interacting with git, see my comment https://github.com/Gandi/gandi.cli/issues/215#issuecomment-314685769

**Comment by sayoun** [comment]
I pushed a fix for tests so they don't rely on external git command, this should cover all case. I don't know if you can use the latest master branch from github to confirm this on your side, on Travis all tests are green again.

**Comment by sayoun** [comment]
This should be fixed in 0.22, I'm closing this, feel free to reopen if this is still an issue.
